File: 1913-maximum-product-difference-between-two-pairs/1913-maximum-product-difference-between-two-pairs.cpp
class Solution {
public:
    int maxProductDifference(vector<int>& nums) {
        int v1=0, v2=0, v3= INT_MAX, v4= INT_MAX;
        for(int i=0; i<nums.size(); i++){
            if(v1<nums[i]){
                v2=v1;
                v1=nums[i];
            }else if(nums[i]<=v1 && v2<nums[i]) v2= nums[i];
            
            if(v3> nums[i]){
                v4=v3;
                v3= nums[i];
            }else if(nums[i]>= v3 && v4>nums[i]) v4= nums[i];
        }
        return v2*v1 - v3*v4;
    }
};

File: 1913-maximum-product-difference-between-two-pairs/1913-maximum-product-difference-between-two-pairs.java
class Solution {
    public int maxProductDifference(int[] nums) {
        int maxi=0, smaxi=0, mini=Integer.MAX_VALUE, smini= Integer.MAX_VALUE;
        for(int i=0; i<nums.length; i++){
            if(maxi< nums[i]){
                smaxi= maxi;
                maxi= nums[i];
            }else if(smaxi< nums[i]) smaxi= nums[i];
            if(mini > nums[i]){
                smini= mini;
                mini = nums[i];
            }else if(smini> nums[i]) smini = nums[i];
        }
        
        return ((maxi*smaxi) - (mini *smini));
    }
}